### DISCUSSION TITLE
How to avoid React warning-keys?

### DISCUSSION THREAD

**Discussion by toiroakr** [gqty-dev/gqty#2036] [Q&A] [answered]
The following code will throw an error saying that there is no unique key for the div.
How to avoid the error?
```tsx
export default function Home() {
  const query = useQuery();
  return (
    <div>
      {query.allFilms()?.films?.map((film) => (
        <div key={film?.id}>
          <p>title: {film?.title}</p>
          <p>director: {film?.director}</p>
        </div>
      ))}
    </div>
  );
}
```
[Full code is here.](https://github.com/toiroakr/gqty-key-error)

error:
```
Each child in a list should have a unique "key" prop.
Check the render method of `Home`. See https://react.dev/link/warning-keys for more information.
```

![CleanShot 2024-11-15 at 23 25 44](https://github.com/user-attachments/assets/eb208ee5-2686-496c-88d0-e030659cbb49)

**Comment by vicary** [comment, marked as answer]
GQty renders a first time with `undefined` to capture your GraphQL selections before fetching, assuming no sane person would use `0` as an id you may always to this:

```diff
<div>
  {query.allFirms()?.films?.map((film) => (
-   <div key={film?.id}>
+   <div key={film?.id ?? 0}>
    </div>
  )}
</div>
```

**Reply by toiroakr** [reply]
@vicary 
Thanks! It works.
